WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayInformationViewModel.swift

```diff
@@ -60,12 +60,28 @@ final class SetUpTapToPayInformationViewModel: PaymentSettingsFlowPresentedViewM
             guard let self = self else {
                 return
             }
+            self.disconnectFromBluetoothReader(in: readers)
             self.noConnectedReader = readers.isEmpty ? .isTrue : .isFalse
             self.reevaluateShouldShow()
         }
         stores.dispatch(connectedAction)
     }
 
+    /// This screen is only used for setting up the Built In card reader.
+    /// If we're connected to a Bluetooth reader when the screen opens,
+    /// we should disconnect, as we can't continue with setup while connected.
+    private func disconnectFromBluetoothReader(in readers: [CardReader]) {
+        if readers.includesBluetoothReader() {
+            self.connectionAnalyticsTracker.automaticallyDisconnectedFromReader()
+            self.disconnect()
+        }
+    }
+
+    private func disconnect() {
+        let action = CardPresentPaymentAction.disconnect { _ in }
+        ServiceLocator.stores.dispatch(action)
```

Comment:
Do we need to call the dispatcher from the ServiceLocator here?

```suggestion
        stores.dispatch(action)
```